sagemaker/launch/launch_realtime_endpoint.py

```diff
@@ -0,0 +1,269 @@
+             and model tar file are located!")
+    realtime_infer_parser.add_argument("--instance-type", type=str, default="ml.c6i.xlarge",
+        help="instance type for the SageMaker job")
+    realtime_infer_parser.add_argument("--instance-count", type=int, default=1,
+        help="number of instances")
+
+    # real-time task specific arguments
+    realtime_infer_parser.add_argument("--restore-model-path", type=str,
+        help="The folder path where trained GraphStorm model parameters were saved.")
+    realtime_infer_parser.add_argument("--model-yaml-config-file", type=str,
+        help="The file path to the new YAML configuration file generated in GraphStorm \
+              model training.")
+    realtime_infer_parser.add_argument("--graph-json-config-file", type=str,
+        help="The file path to the updated JSON configuration file created in GraphStorm \
+              graph construction process.")
+    realtime_infer_parser.add_argument("--upload-tarfile-s3", type=str,
+        help="The S3 location for uploading the packed model artifacts tar file.")
+    realtime_infer_parser.add_argument("--infer-task-type", type=str,
+        choices=SUPPORTED_REALTIME_INFER_TASKS,
+        help=(f"The name of real time inference task. Options \
+               include {SUPPORTED_REALTIME_INFER_TASKS}"))
+    realtime_infer_parser.add_argument("--model-name", type=check_name_format,
+        default='GSF-Model4Realtime',
+        help="The name for the to-be created SageMaker objects. The name should follow \
+              a regular expression pattern: ^[a-zA-Z0-9]([\-a-zA-Z0-9]*[a-zA-Z0-9])$. Default \
+              is \"GSF-Model4Realtime\".")
+
+    # customized model specific arguments
+    realtime_infer_parser.add_argument("--model-tarfile-s3", type=str,
+        help="The S3 location of the compressed model tar file. If provided, it will be used \
+              to create a SageMaker Model.")
+    realtime_infer_parser.add_argument("--entrypoint-file-name", type=str,
+        help="The name of the model entry point file. This file name will be used when you \
+              sepcify the --model-tarfile-s3 argument to use a pre-uploaded model tar file.")
+
+    return realtime_infer_parser
+
+def validate_realtime_infer_arguments(args):
+    """ Validate the input arguments with the designed launch logic
+    
+    The real-time inference endpoint lauch has the logic as the followings.
+    
+    1. If users provide the --model-tarfile-s3, it means it is a custom model. Then users must
+       provide the --entrypoint-file-name too, which means users need to implement their own
+       SageMaker entry point file, pack it with other model artifacts in to a compressed
+       tar file, and upload to the given S3 url.
+    2. If users want to use the default launch functions, they must provide five arguments:
+       --restore-model-path
```

Comment:
Don't we have a tool to help users to pack the tar files?
Shall we force the user to upload the tar file to S3 before calling launch_realtime_endpoint?

Reply:
There is a function in the `launch_utils.py`, called `wrap_model_artifacts`, that can be used to pack and zip the tar file. We can later on add a runnable script under the `graphstorm/tools` folder.

For this question, `Shall we force the user to upload the tar file to S3 before calling launch_realtime_endpoint?`, we had a discussion before. The conclusion was to keep the flexibility, that is, we provide script to do automatic packing and deployment, and to allow users to do it separately. The current launch script is based on this conclusion.